src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -23,6 +23,12 @@ private static Type GuessKind(int col, List<string[]> read)
                     throw new FormatException(string.Format(Strings.LessColumnsThatExpected, nbline + 1));
 
                 string val = line[col];
+
+                if (val.Equals("null", StringComparison.OrdinalIgnoreCase))
```

Comment:
Are you guaranteed `val` is never actually `null` here? If it is, you will get a `NullReferenceException`. Instead, it is usually safer to say `string.Equals(val, "null", StringComparison.OrdinalIgnoreCase)`

Reply:
Ahh, good catch! This is what I actually wanted to suggest 😉 
My bad...Thanks.